stablehlo/tests/infer_stablehlo.mlir

```diff
@@ -693,3 +693,108 @@ func.func @pad_with_negative_inferred_bounds(%arg0: tensor<3x?x?xf16, #stablehlo
   %1 = "hlo_test_infer.get_return_types"(%0) : (tensor<*xf16>) -> tensor<*xindex>
   func.return %1 : tensor<*xindex>
 }
+
+// -----
+
+// CHECK-LABEL: @concat_bounds_c0
```

Comment:
it will be easier to follow if we put all c{0...5} in a single test with multiple dims?

Add one test with unranked as well?

Reply:
"with multiple dims" This is possible for `AddOp` but not for `ConcatenateOp`: each test can only contain one "concat" dim thus there's no way merge them into one big tensor to include every cases.

Unranked case is not related to bounds, right? There is a test here: https://github.com/openxla/stablehlo/blob/main/stablehlo/tests/ops_stablehlo.mlir#L872

Reply:
sorry, I was mistaken on combining the tests. 

yes, it is not related to bounds but we need some test for shape inference as well. Also, let's keep all concatenate ops together. There is `func.func @concat` earlier in this file.

optional: Instead of c0 to c5, consider describing in name or comment about what that particular test is testing. may be like static size with static bound.

Reply:
Sure reviewers. Added the comment of c0~5 rules in test as well and also refer it to `Base.cpp`.

"we need some test for shape inference as well" yes we do. The `concat_1D_unranked` test case is for unranked case. I think you are looking for a test with `hlo_test_infer.get_return_types` and CHECK `type0 = tensor<*x>`. Now I need think about whether we need this (all ops missing this) in later PRs.

"keep all concatenate ops together." It is a good idea. But now every bounded tests are put under spliter:
//===------------===//
// Bounded Dynamism
//===------------===//
So let leave it to future work on how to scatter the tests into proper different files.